sources/packages/backend/libs/test-utils/src/factories/program-year.ts

```diff
@@ -19,3 +26,26 @@ export function createFakeProgramYear(programYearPrefix?: number): ProgramYear {
   programYear.endDate = `${endingYear}-07-31`;
   return programYear;
 }
+
+/**
+ * Ensure that a program year exists, checking by its name.
+ * If it does not exists create a default using {@link createFakeProgramYear}
+ * and save it.
+ * @param db e2e data sources.
+ * @param programYearPrefix program year prefix, for instance, a prefix 2000
+ * would create a program year 2000-2001.
+ * @returns program year with the prefix.
+ */
+export async function ensureProgramYearExists(
+  db: E2EDataSources,
+  programYearPrefix: number,
+): Promise<ProgramYear> {
+  const fakeProgramYear = createFakeProgramYear(programYearPrefix);
+  const existingProgramYear = await db.programYear.findOneBy({
+    programYear: fakeProgramYear.programYear,
+  });
+  if (existingProgramYear) {
+    return existingProgramYear;
+  }
+  return await db.programYear.save(fakeProgramYear);
```

Comment:
do we need to await? nevermind, I think Sonar is not complaining about it

Reply:
Nope, it is a left over. I will remove it.

Reply:
Changed.